packages/lesswrong/components/users/UserContentFeed.tsx

```diff
@@ -0,0 +1,475 @@
+      results {
+        ...CommentsList
+        post {
+          ...PostsListWithVotes
+        }
+        topLevelComment {
+          ...CommentsListWithTopLevelComment
+        }
+      }
+      totalCount
+    }
+  }
+`);
+
+const THREAD_BY_TOPLEVEL_QUERY = gql(`
+  query UserContentFeedThread($topLevelCommentId: String!, $limit: Int) {
+    comments(selector: { repliesToCommentThreadIncludingRoot: { topLevelCommentId: $topLevelCommentId } }, limit: $limit) {
+      results {
+        ...UltraFeedComment
+      }
+    }
+  }
+`);
+
+const styles = defineStyles("UserContentFeed", (theme: ThemeType) => ({
+  root: {
+    display: 'flex',
+    flexDirection: 'column',
+    flex: 1,
+  },
+  sortToggle: {
+    display: 'flex',
+    width: '100%',
+    marginBottom: 8,
+
+  },
+  sortButton: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    fontFamily: theme.typography.fontFamily,
+    fontSize: 14,
+    lineHeight: '20px',
+    fontWeight: 600,
+    padding: '12px 0',
+    width: '50%',
+    cursor: 'pointer',
+    position: 'relative',
+    transition: 'background-color 0.2s',
+    backgroundColor: theme.palette.panelBackground.bannerAdTranslucentHeavy,
+    '&:hover': {
+      backgroundColor: theme.palette.grey[100],
+    },
+  },
+  sortButtonInactive: {
+    color: theme.palette.grey[500],
+  },
+  sortButtonActive: {
+    color: theme.palette.text.primary,
+  },
+  tabLabel: {
+    position: 'relative',
+    display: 'inline-flex',
+    flexDirection: 'column',
+  },
+  tabUnderline: {
+    position: 'absolute',
+    bottom: -12, // Adjust to position at bottom of parent button padding
+    left: 0,
+    right: 0,
+    height: 3,
+    backgroundColor: theme.palette.primary.main,
+    borderRadius: '3px 3px 0 0',
+  },
+  feedContent: {
+    display: 'flex',
+    flexDirection: 'column',
+    marginLeft: 8,
+    marginRight: 8,
+    gap: 8,
+  },
+  loading: {
+    padding: 40,
+    display: 'flex',
+    flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+}));
+
+export function UltraFeedPrefetchedThreadItem({ comment, index, feedSettings }: { 
+  comment: CommentItem; 
+  index: number; 
+  feedSettings: UltraFeedSettingsType; 
+}) {
+  const topLevelId = comment.topLevelCommentId ?? comment._id;
+  const { data: threadData } = useQuery<UserContentFeedThreadQuery, UserContentFeedThreadQueryVariables>(THREAD_BY_TOPLEVEL_QUERY, {
+    variables: { topLevelCommentId: topLevelId, limit: 200 },
+    skip: !topLevelId,
+    notifyOnNetworkStatusChange: false,
+    fetchPolicy: 'cache-first',
+  });
+
+  // Build ancestry chain: results in array of comments from root down to focused comment
+  const ancestryChain = useMemo(() => {
+    const allThreadComments = threadData?.comments?.results ?? [comment];
+    const chain: Array<UltraFeedComment & { __typename?: 'Comment' }> = [];
```

Comment:
Do we need all the `_typename?` fields added to all the type annotations?  What goes wrong if we omit them?